slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1286,15 +1305,73 @@ void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
                 Parallel(&Module::startAcquisition, master);
             }
         }
-    } else {
+    }
+    // all in parallel
+    else {
         if (blocking) {
-            Parallel(&Module::startAndReadAll, positions);
+            Parallel(&Module::startAndReadAll, pos);
         } else {
-            Parallel(&Module::startAcquisition, positions);
+            Parallel(&Module::startAcquisition, pos);
         }
     }
 }
 
+void DetectorImpl::sendSoftwareTrigger(const bool block, Positions pos) {
+    bool handleSynchronization = false;
+    // multi module sync enabled jungfrau
+    if (size() > 1 && (shm()->detType == defs::JUNGFRAU) &&
+        (Parallel(&Module::getSynchronizationFromStopServer, pos)
+             .tsquash("Inconsistent synchronization among modules"))) {
+        handleSynchronization = true;
+    }
+
+    // slaves first
+    if (handleSynchronization) {
+        std::vector<int> master;
+        std::vector<int> slaves;
+        getMasterSlaveList(pos, master, slaves);
+        if (!slaves.empty())
+            Parallel(&Module::sendSoftwareTrigger, slaves, false);
+        if (!master.empty())
+            Parallel(&Module::sendSoftwareTrigger, master, block);
+    }
+    // all in parallel
+    else {
+        Parallel(&Module::sendSoftwareTrigger, pos, block);
+    }
+}
+
+void DetectorImpl::stopDetector(Positions pos) {
```

Comment:
Since we anyway want to stop does the order actually matters?

Reply:
Apparently it does, both @mozzanica and @anberga said so.